doc/roadmap.rst

```diff
@@ -0,0 +1,271 @@
+﻿.. _roadmap:
+
+Roadmap
+=======
+
+Purpose of this document
+------------------------
+This document list general directions that core contributors are interested
+to see developed in scikit-learn. The fact that an item is listed here is in
+no way a promise that it will happen, as resources are limited. Rather, it
+is an indication that help is welcomed on this topic.
+
+Statement of purpose: Scikit-learn in 2018
+------------------------------------------
+Eleven years after the inception of Scikit-learn, much has changed in the
+world of machine learning. Key changes include:
+
+* Computational tools: The exploitation of GPUs, distributed programming
+  frameworks like Scala/Spark, etc.
+* High-level Python libraries for experimentation, processing and data
+  management: Jupyter notebook, Cython, Pandas, Dask, Numba...
+* Changes in the focus of machine learning research: artificial intelligence
+  applications (where input structure is key) with deep learning,
+  representation learning, reinforcement learning, domain transfer, etc.
+
+A more subtle change over the last decade is that, due to changing interests
+in ML, PhD students in machine learning are more likely to contribute to
+PyTorch, Dask, etc. than to Scikit-learn, so our contributor pool is very
+different to a decade ago.
+
+Scikit-learn remains very popular in practice for trying out canonical
+machine learning techniques, particularly for applications in experimental
+science and in data science. A lot of what we provide is now very mature.
+But it can be costly to maintain, and we cannot therefore include arbitrary
+new implementations. Yet Scikit-learn is also essential in defining an API
+framework for the development of interoperable machine learning components
+external to the core library.
+
+**Thus our main goals in this era are to**:
+
+* continue maintaining a high-quality, well-documented collection of canonical
+  tools for data processing and machine learning within the current scope
+  (i.e. rectangular data largely invariant to column and row order;
+  predicting targets with simple structure)
+* improve the ease for users to develop and publish external components
+* improve inter-operability with modern data science tools (e.g. Pandas, Dask)
+  and infrastructures (e.g. distributed processing)
+
+Many of the more fine-grained goals can be found under the `API tag
+<https://github.com/scikit-learn/scikit-learn/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3AAPI>`_
+on the issue tracker.
+
+Architectural / general goals
+-----------------------------
+The list is numbered not as an indication of order of priority, but to make
+referring to specific points easiest. Please add new entries only at the end.
+
+#. Everything in sklearn should conform to our API contract
+
+   * Pipeline and FeatureUnion modify their input parameters in fit. Fixing this

```

Comment:
If you just add single ticks around the estimators, will the become the proper links?